TigranKirakosov/hw4.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+// Class Order
+function Order(customer) {
+  this.customer = customer || 'Officiant :)';
+  this.positions = [];
+}
+
+// Order methods
+Order.prototype.addToOrder = function() {
+  for (var i = 0; i < arguments.length; i++) {
+    this.positions.push(arguments[i]);
+  }
+};
+Order.prototype.deleteFromOrder = function(position) {
+  return (this.positions = this.positions.filter(function(posToKeep) {
+    return posToKeep !== position;
+  }));
+};
+Order.prototype.showPositions = function() {
+  return this.positions
+    .map(function(position) {
+      return position.description;
+    })
+    .join(', ');
+};
+Order.prototype.calcTotalPrice = function() {
+  return `Price is ${this.positions.reduce(function(sum, position) {
+    return sum + position.showPrice();
+  }, 0)} tugrics`;
+};
+Order.prototype.calcTotalCalories = function() {
+  return `Provides ${this.positions.reduce(function(sum, position) {
+    return sum + position.showCalories();
+  }, 0)} calories`;
+};
+Order.prototype.makePayment = function() {
+  return Object.freeze(this), Object.freeze(this.positions);
+};
+
+// Class Food
+function Food(description) {
+  this.description = description || 'Nameless position';
+  this.typeOfFood;
+}
+
+// Food methods
+Food.prototype.chooseType = function(type, menu) {
+  return (this.typeOfFood = menu.filter(function(food) {
+    return food.type === type;
+  })[0]);
+};
+Food.prototype.chooseSize = function(type, menu) {
+  return (this.size = menu.filter(function(size) {
+    return size.type === type;
+  })[0]);
+};
+Food.prototype.chooseAddition = function(type, menu) {
+  return (this.addition = menu.filter(function(addition) {
+    return addition.type === type;
+  })[0]);
+};
+Food.prototype.showAddition = function() {
+  return `${this.addition.type} addition`;
+};
+Food.prototype.showPrice = function() {
+  return this.typeOfFood.tugric;
+};
+Food.prototype.showCalories = function() {
+  return this.typeOfFood.calories;
+};
+Food.prototype.showType = function() {
+  return `It is ${this.typeOfFood.type}`;
+};
+Food.prototype.showSize = function() {
+  return `${this.size.type} size`;
+};
+
+// Class Hamburger
+function Hamburger(description) {
+  Food.call(this, description);
+  this.size;
+  this.addition;
+  this.chooseSize = function(type) {
+    Food.prototype.chooseSize.call(this, type, Hamburger.SIZES);

```

Comment:
I think that `Food.prototype.chooseSize` and `Food.prototype.chooseAddition` are equal to `Food.prototype.chooseType` - you can just reuse it and don't create this methods in `Food`

Reply:
just make 3 parameter that will define which field should set this method and make it `typeOfFood` by default